nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -553,12 +555,8 @@ def fwd_output_and_loss_func(dataloader_iter, model, checkpoint_activations_all_
                                 else None
                             )
                 else:
-                    # Intermediate pipeline stage only needs attention_mask
-                    if self.get_attention_mask_from_fusion:
-                        batch = {k: None for k in ['tokens', 'position_ids', 'attention_mask', 'labels']}
-                    else:
-                        for k in batch.keys():
-                            batch[k] = batch[k].cuda(non_blocking=True) if k in ['attention_mask'] else None
+                    # Intermediate pipeline stage doesn't need any inputs
+                    batch = {k: None for k in ['tokens', 'position_ids', 'attention_mask', 'labels']}
```

Comment:
I'm uncertain how to handle this case. If we want to keep the same behavior, we need to access the data loader in the case `self.get_attention_mask_from_fusion==False` so that we can access `attention_mask`. But that will cause a hang. Do we just require `self.get_attention_mask_from_fusion==True` to support interleaved pipeline parallelism?

Reply:
I'm fine with this. Can you also default get_attention_mask_from_fusion to True in the megatron gpt config and the model?